source/hash_table.c

```diff
@@ -646,6 +646,37 @@ int aws_hash_table_foreach(
     return AWS_OP_SUCCESS;
 }
 
+bool aws_hash_table_eq(
+    bool (*value_eq)(const void *a, const void *b),
+    const struct aws_hash_table *a,
+    const struct aws_hash_table *b) {
+    if (aws_hash_table_get_entry_count(a) != aws_hash_table_get_entry_count(b)) {
+        return false;
+    }
+
```

Comment:
If, as the docs say, the key comparators must match, shouldn't that be verified here by comparing equals_fn on both?

Reply:
We don't require that they are _equal_, just that they are _equivalent_. For example, we might pass in two FFI-generated bindings to the same lambda function in an interpreted language.